core/mm/core_mmu.c

```diff
@@ -2524,17 +2541,22 @@ static void *phys_to_virt_tee_ram(paddr_t pa, size_t len)
 	 */
 	va = map_pa2va(mmap, pa, len);
 
-	if (va && mmap->type == MEM_AREA_TEE_RAM_RW) {
+	if (mmap->type == MEM_AREA_TEE_RAM_RW && vcore_free_mapped_pages &&
```

Comment:
Add a test on `va` or `mmap` in case all `find_map_by_type_and_pa()` calls failed.

Maybe we should add tests in **core/pta/test/misc.c** on `phys_to_virt()` and `virt_to_phys()` for few valid and invalid translation cases. It would have find this issue.

Reply:
Good idea, I'll add a few.